core/src/main/scala/org/apache/spark/sql/SnappyParser.scala

```diff
@@ -663,7 +673,7 @@ class SnappyParser(session: SnappySession)
     SELECT ~ (DISTINCT ~> trueFn).? ~
     (namedExpression + commaSep) ~
     (FROM ~ ws ~ relations).? ~
-    (WHERE ~ expression).? ~
+    (WHERE ~ (test(setTokenize) ~ expression ~ test(setNoTokenize))).? ~
```

Comment:
Perhaps all-caps names for the rules like TOKENIZE_START and TOKENIZE_END to make them stand out.